osu.Game/Beatmaps/BeatmapMetadata.cs

```diff
@@ -128,5 +129,7 @@ public bool Equals(BeatmapMetadata other)
                    && AudioFile == other.AudioFile
                    && BackgroundFile == other.BackgroundFile;
         }
+
+        public BeatmapMetadata DeepClone() => (BeatmapMetadata)MemberwiseClone();
```

Comment:
I'm not sure if this is what we want. It means that individual difficulties can have differing artist/titles, which while technically is supported by stable and by the lazer database structure, is usually not what a mapper wants:

https://user-images.githubusercontent.com/191335/133973245-babff722-2622-4301-aa4d-97267c8295f1.mp4

A good initial starting point might be to drop this and just have all difficulties share metadata until we have a better UX flow for exceptional cases in place.